build/build.proj

```diff
@@ -334,6 +334,39 @@
     </MsBuild>
   </Target>
 
+  <!--
+    ============================================================
+    Get projects need to be packed
+    ============================================================
+  -->
+  <Target Name="GetPackProjects">
+      <MsBuild
+        Projects="@(SolutionProjects)"
+        Targets="GetPackProject">
+          <Output
+              TaskParameter="TargetOutputs"
+              ItemName="PackProjects" />
+      </MsBuild>
+  </Target>
+
+  <!--
+    ============================================================
+    Ensure all the nupkg and symbol.nugkg files created
+    ============================================================
+  -->
+  <Target Name="EnsurePackagesExist" DependsOnTargets="GetPackProjects">
+      <ItemGroup>
+          <NupkgsFiles Include="$(NupkgOutputDirectory)\*.nupkg"
+                       Exclude="$(NupkgOutputDirectory)\*.symbols.nupkg" />
+          <SymbolFiles Include="$(NupkgOutputDirectory)\*.symbols.nupkg" />
+      </ItemGroup>
+
+      <Message Text="Count of Projects need to be packed: @(PackProjects->Count())        Nupkg Count: @(NupkgsFiles->Count())        Symbol Count: @(SymbolFiles->Count())" Importance="high" />
```

Comment:
I think it would've been nicer if we validated the *exact* package ids.

Reply:
I understand that will be more work, more complicated, so get a 2nd opinion before jumping into it. It's an idea, not necessarily a request for change.

Reply:
When I created the issue, I had imagined a powershell script that ran after pack that checked the filenames or package ids or something. A separate list of expected packages in any case.

This implementation avoids having redundant data to check that all the packages were created successfully, that needs to be maintained each time we add/remove packages. But a redundant list would provide an additional fail-safe. We very rarely add/remove packages, so it wouldn't be a big burden. This change has a theoretical risk that if a project accidentally stops being packable for any reason (and since it's via msbuild, it's a non-trivial risk due to automatic imports, environment variables, etc) we won't catch it.

Having said that, we also have a task to modify our publishing pipeline to validate all the packages exist before we try to push. The problem there is if we detect an error, we have to spin a new build.

Overall, I'm undecided. On one hand, I did imagine us having a list of package/file names that we expected to exist. On the other hand, I believe this change would have caught the problem that prevented our 5.8 packages from being generated correctly, and the theoretical risk of accidentally setting IsPackable to something other than true is low risk.

The more I think about it, the more fine I am, personally.

Reply:
My understanding is, [pack task](https://github.com/NuGet/NuGet.Client/blob/dev/build/templates/Build_and_UnitTest.yml#L209-L216) runs [pack target](https://github.com/NuGet/NuGet.Client/blob/dev/build/build.proj#L221-L229), which calls [PackProjects](https://github.com/NuGet/NuGet.Client/blob/dev/build/common.targets#L137) target. And the PackProjects target only runs when `PackProject` is true.
So the count of projects with `PackProject= true` should be the count of packages need to be created. If the count of `*.nupkg/*.symbols.nupkg` files both match with the count of projects need to be packed, the only risk is there is some old or other `*.nupkg/*.symbols.nupkg` in the folder, making the count exactly match(miss packing certain packages, but has exactly numbers of other packages, so the counts still match), which has a very low chance when running on CI.
I'll add a checklist to check for exact package ids in another PR later.

Reply:
> Overall, I'm undecided. On one hand, I did imagine us having a list of package/file names that we expected to exist. On the other hand, I believe this change would have caught the problem that prevented our 5.8 packages from being generated correctly, and the theoretical risk of accidentally setting IsPackable to something other than true is low risk.


I think either way is good enough as well.

Reply:
I've added checking on each package ids. @nkolev92  @zivkan

Reply:
Reverted the last commit as it doesn't provide any additional safety.